sunpy/physics/differential_rotation.py

```diff
@@ -227,102 +408,162 @@ def _warp_sun_coordinates(xy, smap, dt: u.s, **diffrot_kwargs):
     else:
         smap_data = smap.data
 
-    submap = False
-    # Check whether the input is a submap
-    if ((2 * smap.rsun_obs > smap.top_right_coord.Tx - smap.bottom_left_coord.Tx) or
-        (2 * smap.rsun_obs > smap.top_right_coord.Ty - smap.bottom_left_coord.Ty)):
-
-        submap = True
-        if pad:
-            # Calculating the largest distance between the corners and their rotation values
-            deltax = deltay = 0
-            for corner in product(*product([0 * u.pix], smap.dimensions)):
-                corner_world = smap.pixel_to_world(*corner)
-                corner_world_rotated = solar_rotate_coordinate(corner_world, new_time, **diffrot_kwargs)
-                corner_px_rotated = smap.world_to_pixel(corner_world_rotated)
-                dx = np.abs(corner_px_rotated.x - corner[0])
-                dy = np.abs(corner_px_rotated.y - corner[1])
-                deltax = dx if dx > deltax else deltax
-                deltay = dy if dy > deltay else deltay
-
-            deltax = np.int(np.ceil(deltax.value))
-            deltay = np.int(np.ceil(deltay.value))
-            # Create a new `smap` with the padding around it
-            smap_data = np.pad(smap.data, ((deltay, deltay), (deltax, deltax)),
-                               'constant', constant_values=0)
-            smap_meta = deepcopy(smap.meta)
-            smap_meta['naxis2'], smap_meta['naxis1'] = smap_data.shape
-            smap_meta['crpix1'] += deltax
-            smap_meta['crpix2'] += deltay
-            smap = sunpy.map.Map(smap_data, smap_meta)
-
-    warp_args = {'smap': smap, 'dt': dt}
-    warp_args.update(diffrot_kwargs)
+    # Create the arguments for the warp function.
+    warp_args = {'smap': smap, 'new_observer': new_observer}
+    warp_args.update(diff_rot_kwargs)
+
     # Apply solar differential rotation as a scikit-image warp
-    out = transform.warp(to_norm(smap_data), inverse_map=_warp_sun_coordinates,
-                         map_args=warp_args)
+    out_data = transform.warp(to_norm(smap_data), inverse_map=_warp_sun_coordinates, map_args=warp_args)
 
     # Recover the original intensity range.
-    out = un_norm(out, smap.data)
+    out_data = un_norm(out_data, smap.data)
 
-    # Update the meta information with the new date and time, and reference pixel.
+    # Update the meta information with the new date and time.
     out_meta = deepcopy(smap.meta)
     if out_meta.get('date_obs', False):
         del out_meta['date_obs']
-    out_meta['date-obs'] = "{}".format(new_time.strftime('%Y-%m-%dT%H:%M:%S'))
-
-    if submap:
-        crval_rotated = solar_rotate_coordinate(smap.reference_coordinate, new_time, **diffrot_kwargs)
-        out_meta['crval1'] = crval_rotated.Tx.value
-        out_meta['crval2'] = crval_rotated.Ty.value
-
-    return sunpy.map.Map((out, out_meta))
+    out_meta['date-obs'] = new_observer.obstime.strftime("%Y-%m-%dT%H:%M:%S.%f")
+
+    # Need to update the observer location for the output map.
```

Comment:
post #3115  this should fix the test fails:
```suggestion
    # Need to update the observer location for the output map.
    from sunpy.util import expand_list  # move me to the top
    all_keys = expand_list([e[0] for e in smap._supported_observer_coordinates])
    for key in all_keys:
        out_meta.pop(key)
```